frontend/test/metabase/scenarios/question/settings.cy.spec.js

```diff
@@ -130,18 +130,48 @@ describe("scenarios > question > settings", () => {
       cy.findByText("Visible columns").click();
       findColumnAtIndex("Products → Category", 5);
 
+      // https://github.com/metabase/metabase/pull/21338#pullrequestreview-928807257
+
+      // Add "Address"
+      cy.findByText("Address")
+        .siblings(".Icon-add")
+        .click();
       /**
        * Helper functions related to THIS test only
        */
 
+      // The result automatically load when adding new fields
+      cy.wait("@dataset");
```

Comment:
@flamber I've noticed something strange in the Metabase at this point. It fires **two** `POST /api/dataset` xhrs. One seems to be related to the reordering (adding the new column), and another one is actually returning the query. Seems a bit wasteful - not sure if that was intended or a code smell that have to be looked at?

Reply:
Yeah, as @WiNloSt also mentioned (also mentioned in the issue), it runs the query and actually shows the change, but then it shows the Run-overlay, which causes double execution.
I simply don't know how our code is structured and if it can easily be prevented, but we can open another issue for that after this has been fixed.

Reply:
@flamber Just tested. It seems anytime you add a new field it calls `/api/dataset` twice. And when you add a new field it doesn't even show the run-overlay (the "Play button"), it seems that's automatic. I'll see if there's a quick win.